web/services/HolidayService.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/*
+ * Copyright (C) 2021 Igalia, S.L. <info@igalia.com>
+ *
+ * This file is part of PhpReport.
+ *
+ * PhpReport is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PhpReport is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PhpReport.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Phpreport\Web\services;
+
+if (!defined('PHPREPORT_ROOT')) define('PHPREPORT_ROOT', __DIR__ . '/../../');
+
+include_once(PHPREPORT_ROOT . '/web/services/WebServicesFunctions.php');
+include_once(PHPREPORT_ROOT . '/model/facade/UsersFacade.php');
+include_once(PHPREPORT_ROOT . '/model/vo/UserVO.php');
+require_once(PHPREPORT_ROOT . '/util/LoginManager.php');
+
+class HolidayService
+{
+    private \LoginManager $loginManager;
+
+    public function __construct(
+        \LoginManager $loginManager
+    ) {
+        $this->loginManager = $loginManager;
+    }
+
+    public function datesToRanges(array $vacations): array
```

Comment:
@jaragunde I tried to make this method as simple as possible but I know it's still a bit confusing, let me know if you can think of a better way to implement this :)

Reply:
I've added a description with examples on top of it, let me know  if it's better